### PR TITLE
fix: update formula to calculate tvl of sicx

### PR DESCRIPTION
**Pull request by shallcdn**
**Twitter Link: https://twitter.com/helloiconworld**

**List of audit links if any:**

**Website Link: https://icon.community/**

**Logo (High resolution, preferably in .svg and .png, for application on both white and black backgrounds. Will be shown with rounded borders): https://bit.ly/3OdMrui**

**Current TVL: $60,78 M**

**Chain: ICON**

**Coingecko ID (so your TVL can appear on Coingecko): (https://api.coingecko.com/api/v3/coins/list) staked-icx**

**Coinmarketcap ID (so your TVL can appear on Coinmarketcap): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000) N/A**

**Short Description (to be shown on DefiLlama): sICX is a token that represents a share of a pool of staked ICX, so it’s always accruing staking rewards and compounding automatically.**

**Token address and ticker if any: sICX - cx2609b924e33ef00b648a409245c7ea394c467824**

**Category (full list at https://defillama.com/categories): Staking**

**Oracle used (Chainlink/Band/API3/TWAP or any other that you are using): Band**

**forkedFrom (Does your project originate from another project): No**

**methodology (what is being counted as tvl, how is tvl being calculated): Gets the totalClaimableIcx value from API (the total amount of ICX held by the contract), then multiplies it by the ICX/USD price.**

### CODE REVIEW

**Comment by shallcdn** [comment]
Screen Shot 2022-04-26 at 23 06 35

**Comment by llamatester** [comment]
The adapter at projects/sicx exports TVL: 
        
 
 
    icon                      61.05 M
    
    total                    61.05 M

File: projects/sicx/index.js [hunk 1]
```diff
@@ -2,27 +2,56 @@ const axios = require("axios")
...
 async function fetch() {
-    let totalClaimableIcx = await getTotalClaimableIcx()
-    return { 'icon': totalClaimableIcx }
+
+    let icxPrice = await getIcxUsdPrice();
+    let totalClaimableIcx = await getTotalClaimableIcx();
+    let totalSupply = await getTotalSupply();
+    let todayRate = await getTodayRate();
+    // tvl = ICX/USD Price * (totalClaimableIcx + totalSupply from sICX token SCORE * getTodayRate)
+    const tvl = icxPrice * (totalClaimableIcx + totalSupply * todayRate);
```
Comment (g1nt0ki) [inline projects/sicx/index.js:53]: I think this calculation is bit off, we are returning amount of icon tokens here, and we later fetch the price from coingecko, so no need to fetch the price

**Comment by llamatester** [comment]
Error while running adapter at projects/sicx: 
        
 
  
    
    
    ReferenceError: getIcxUsdPrice is not defined
        at fetch (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/sicx/index.js:36:20)
        at getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:57:31)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:149:15
        at Array.map (<anonymous>)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:137:29
        at Array.map (<anonymous>)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:132:44
        at processTicksAndRejections (node:internal/process/task_queues:96:5)

**Comment by llamatester** [comment]
The adapter at projects/sicx exports TVL: 
        
 
 
    icon                      76.63 M
    
    total                    76.63 M

File: projects/sicx/index.js (hunk 1 above)
Comment (shallcdn) [reply projects/sicx/index.js]: Hi @g1nt0ki, we the calculation should be `totalClaimableIcx + totalSupply * todayRate`. Please help to update it. Thank you

Comment (g1nt0ki) [reply projects/sicx/index.js]: @shallcdn I am a bit confused, wouldnt `totalSupply * todayRate` include the claimable icx as well? I see no option for end user who converted ICX -> sICX to claim anything

Comment (g1nt0ki) [reply projects/sicx/index.js]: > @g1nt0ki Let me explain a bit. As you see in the code we have 2 contract:
> 
> **sICX token contract** = cx2609b924e33ef00b648a409245c7ea394c467824 **Staking Management contract** = cx43e2eec79eb76293c298f2b17aec06097be606e0
> 
> `totalSupply` got from **sICX token contract** it means total sICX already minted `totalClaimableIcx` got from **Staking Management contract** it means total ICX already staked in this contract,
> 
> Further, the price of sICX is higher than ICX, so we have `todayRate` to get the rate between ICX and sICX and then multiply `totalSupply`
> 
> As we know, sICX is staked ICX, so to the tvl of sICX will be calculated: `ICX/USD Price * (totalClaimableIcx + totalSupply from sICX token SCORE * getTodayRate`

I understand, but my point is, I am seeing totalClaimableIcx (which is staked ICX?) this quantity is way below sICX total supply, is the rest of ICX is in some other contract for staking?

if I do `totalSupply * getTodayRate`, wont it give me all the ICX staked + rewards accumulated ? if so, shouldnt we leave out `totalClaimableIcx` from tvl calculation?

**Pull request merged**